plugins/vpc-eni/config/netconfig.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+
+	"github.com/aws/amazon-vpc-cni-plugins/network/vpc"
+	log "github.com/cihub/seelog"
+
+	cniSkel "github.com/containernetworking/cni/pkg/skel"
+	cniTypes "github.com/containernetworking/cni/pkg/types"
+)
+
+// NetConfig defines the network configuration for the vpc-eni plugin.
+type NetConfig struct {
+	cniTypes.NetConf
+	ENIName            string
+	ENIMACAddress      net.HardwareAddr
+	ENIIPAddress       *net.IPNet
+	GatewayIPAddress   net.IP
+	NoInfraContainer   bool
+	UseExistingNetwork bool
+}
+
+// netConfigJSON defines the network configuration JSON file format for the vpc-eni plugin.
+type netConfigJSON struct {
```

Comment:
The vpc-eni netconfig json format should be wire-compatible with that of ecs-eni's. For example, ecs-eni has an array of IP addresses (v4 and v6). Even if vpc-eni won't support multiple IP addresses, we should still define this as an array with a single element. Comment applies to all fields.

Reply:
Sure, I'll make the common fields between this plugin and `ecs-eni` plugin cross-compatible.

Reply:
We've decided to use the vpc-shared-eni naming offline.